src/emulate.c

```diff
@@ -1201,6 +1272,18 @@ riscv_word_t rv_get_reg(struct riscv_t *rv, uint32_t reg)
     return ~0U;
 }
 
+/* allocate a new code block */
+struct block_t *rv_block_alloc()
```

Comment:
Since we are going to perform memory allocation frequently, the technique [memory pool](https://en.wikipedia.org/wiki/Memory_pool) can be applied. Check https://github.com/jonas-fan/mempool for a simplified example. My tests:
```
# Test 001 
os malloc/free ... elapsed 246.09ms
mempool alloc/free ... elapsed 8.72ms
# Test 002 
os malloc/free ... elapsed 658.55ms
mempool alloc/free ... elapsed 24.38ms
# Test 003 
os malloc/free ... elapsed 483.08ms
mempool alloc/free ... elapsed 21.50ms
```

Reply:
I think that I can use queue to manage memory block. Rewrite as the following:

```c
/* a translated basic block */
struct block_t {
    /* memory blocks */
    struct rv_insn_t *ir;
    /* index of memory blocks */
    uint32_t front, rear;
    /* maximum of instructions encompased */
    uint32_t capacity;
};
```

That is, "decode" means enqueue a memory block, and "execute" means dequeue a memory block.